yage-core/src/core/base_animation.rs

```
@@ -0,0 +1,200 @@
+/// and value ranges, see [`Animation`].
+///
+/// [`Animation`]: struct.Animation.html
+///
+pub struct BaseAnimation {
+    duration: f64, // Number of seconds between 0.0 and 1.0
+    looped: bool, // Is the animation looped?
+    bouncing: bool, // Will the animation go back and forth?
+    running: bool, // Current status
+    back: bool, // Is the animation on its way back? (in case of bouncing)
+    value: f64 // Current value ([0..1])
+}
+
+impl BaseAnimation {
+    ///
+    /// Create animation.
+    ///
+    /// # Returns
+    /// A new instance of BaseAnimation.
+    ///
+    pub fn new() -> Self {
+        Self {
+            duration: 1.0,
+            looped: false,
+            bouncing: false,
+            running: false,
+            back: false,
+            value: 0.0
+        }
+    }
+
+    ///
+    /// Create animation with options.
+    ///
+    /// # Parameters
+    /// - `duration`: Duration (in seconds)
+    /// - `looped`: true if the animation is looped, else false
+    /// - `bouncing`: true if the animation is bouncing, else false
+    /// - `start`: true to start the animation right away, else false
+    ///
+    /// # Returns
+    /// A new instance of BaseAnimation.
+    ///
+    pub fn with_options(duration: f64, looped: bool, bouncing: bool, start: bool) -> Self {
+        Self {
+            duration: duration,
+            looped,
+            bouncing,
+            running: start,
+            back: false,
+            value: 0.0
+        }
+    }
+
+    ///
+    /// Get current value.
+    ///
+    /// # Returns
+    /// Value ([0.0 .. 1.0])
+    ///
+    pub fn get_value(&self) -> f64 {
+        self.value
+    }
+}
+
+impl Animate for BaseAnimation {
+    fn get_duration(&self) -> f64 {
+        self.duration
+    }
+
+    fn set_duration(&mut self, duration: f64) {
+        // Do not accept empty or negative duration
+        if duration > 0.0 {
+            self.duration = duration;
+        }
```

How about a log or a panic? I don't quite like silently ignoring bad input.

---

panic is too much. But log definitely